test/evergreen/code_complexity_analysis.py

```diff
@@ -0,0 +1,126 @@
+    ''' Retrieve 3 set of cyclomatic complexity ranges, above 30, above 50 and above 90. '''
+    rangesList = [20, 50, 90]
+    atlasFormat = {}
+    for i in rangesList:
+        aboveRangeString = 'Above ' + str(i)
+        atlasFormat[aboveRangeString] = get_complexity_max_limit(dataFile, i)
+
+    return (atlasFormat)
+
+def get_complexity_max_limit(complexity_data_path: str, max_limit:int):
+    ''' Retrieve the number of cyclomatic complexity values above the max_limit. '''
+    columnName = 'std.code.complexity:cyclomatic'
+    # Select column 'std.code.complexity:cyclomatic' from the dataframe
+    column = dataFrame[columnName]
+    # Get count of values greater than max_limit in the column 'std.code.complexity:cyclomatic'
+    count = column[column > max_limit].count()
+    """
+    Pandas aggregation functions (like sum, count and mean) returns a NumPy int64 type number not a Python integer.
+    Object of type int64 is not JSON serializable so need to convert into an int.
+    """
+    return (int(count))
+
+def get_average(complexity_summary_file):
+    ''' Retrieve the cyclomatic code complexity value '''
+    with open(complexity_summary_file, 'r') as f:
+        data = f.read()
+
+    # Below is the format of complexity_summary_file, we are intersted in extracting the "avg" value of std.code.complexity.
+    """
+    {
+        "view": [
+            {
+                "data": {
+                    "info": {"path": "./", "id": 1},
+                    "aggregated-data": {
+                        "std.code.complexity": {
+                            "cyclomatic": {
+                                "max": 91,
+                                "min": 0,
+                                "avg": 4.942372881355932,
+                                "total": 14580.0,
+                                "count": 2950,
+                                "nonzero": False,
+                                "distribution-bars": [ ],
+                                "sup": 0,
+                            }
+                        },
+                        "std.code.lines": {
+                            "code": { }
+                        },
+                    },
+                    "file-data": {},
+                    "subdirs": [ ],
+                    "subfiles": [ ],
+                }
+            }
+        ]
+    }
+    """
+
+    d = ast.literal_eval(data)
+    newDictionary = d["view"]
+
+    return (newDictionary[0]['data']['aggregated-data']['std.code.complexity']['cyclomatic']['avg'])
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-s', '--summary', required=True, help='Path of the complexity summary file in the json format')
+    parser.add_argument('-o', '--outfile', help='Path of the file to write analysis output to')
+    parser.add_argument('-d', '--data_file', help='Code complexity data file in the csv format')
+
+    args = parser.parse_args()
+    global dataFrame
+    global topNRegions
```

Comment:
Is it necessary to use global variables? 
Looks like the dataFrame variable is red in 2 x functions, can we move lines 121-122 closer to where it's directly used e.g. within `get_code_complexity()` and pass it as an arg for the reader functions?

Reply:
I moved the initializations of the `dataFrame ` and `topNRegions` to the `get_code_complexity` method but still kept it global instead to pass it an argument.

Reply:
There are some caveats about using global variables in Python. I'd rather avoid using them unless necessary. Cann't you pass the 2 variables as function args to the functions that make use of them (e.g. `get_code_complexity`) and make them non-global?

Reply:
Fixed in 19d2a584b833bc89692c079e17033ea0c4609175. Removed using global variables and passing it as an function argument.